test/core/surface/BUILD

```diff
@@ -123,6 +123,7 @@ grpc_cc_test(
     deps = [
         "//:gpr",
         "//:grpc",
+        "//:grpc_authorization_provider",
```

Comment:
@markdroth 
On adding the APIs to grpc_security.h, and then running the generate_projects script, it updates public_headers_must_be_c89 file
https://github.com/grpc/grpc/blob/master/test/core/surface/public_headers_must_be_c89.c
Without adding this target here, we get error like
"undefined reference to grpc_authorization_policy_provider_static_data_create" as it cannot locate the definition. Similarly for release API.
Since grpc_authorization_provider has dependency on RE2, we may not want to do this.

Also, when we add the Setter for GRPC_ARG_AUTHORIZATION_POLICY_PROVIDER in ServerBuilder, we would have to include grpc++_authorization_provider target (which pulls in dependency on RE2) to grpc++_base, something we were trying to avoid.
https://github.com/grpc/grpc/blob/master/include/grpcpp/server_builder.h

Please let me know if I am missing anything. Because to me, It looks like the current BUILD structure may not work.